server/datastore/mysql/software.go

```diff
@@ -3244,6 +3244,16 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 			// data from the byVPPAdamID to hostVPPInstalledTitles
 			// so we can later push to InstalledVersions
 			installedTitle := byVPPAdamID[*s.VPPAppAdamID]
+			if installedTitle == nil {
+				// This can happen when mdm_enrolled is false
+				// because in hostVPPInstalls we filter those out
+				installedTitle = bySoftwareTitleID[s.ID]
+			}
+			if installedTitle == nil {
+				// We somehow have a vpp app in host_vpp_software_installs,
+				// however osquery didnt pick it up in inventory
```

Comment:
```suggestion
				// however osquery didn't pick it up in inventory
```

This would happen if a VPP app was installed but we haven't yet refetched host vitals, right?